dill/_dill.py

```diff
@@ -504,29 +505,138 @@ def dump_session(filename='/tmp/session.pkl', main=None, byref=False, **kwds):
         pickler._main_modified = main is not pickler._original_main
         pickler.dump(main)
     finally:
-        if f is not filename:  # If newly opened file
-            f.close()
+        if file is not filename:  # if newly opened file
+            file.close()
     return
 
-def load_session(filename='/tmp/session.pkl', main=None, **kwds):
-    """update the __main__ module with the state from the session file"""
-    if main is None: main = _main_module
+def _open_peekable(filename):
     if hasattr(filename, 'read'):
-        f = filename
+        file = filename
     else:
-        f = open(filename, 'rb')
-    try: #FIXME: dill.settings are disabled
-        unpickler = Unpickler(f, **kwds)
-        unpickler._main = main
+        file = open(filename, 'rb')
+    if not hasattr(file, 'peek'):
+        try:
+            import io
+            file = io.BufferedReader(file)
+        except Exception:
+            pass  # ...and hope for the best
+    return file
+
+def _inspect_pickle(file, main_is_none):
```

Comment:
This function seems badly named. Can you adjust to better repent the result of the function?